x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -70,12 +88,15 @@ export class InnerJoin {
           featurePropertyKey.length >= stylePropertyPrefix.length &&
           featurePropertyKey.substring(0, stylePropertyPrefix.length) === stylePropertyPrefix
         ) {
-          delete feature.properties[featurePropertyKey];
+          // For some reason, ts compilation complains about this deletion
+          if (feature.properties) {
+            delete feature.properties[featurePropertyKey];
+          }
         }
       });
     }
 
-    const joinKey = feature.properties[this._leftField.getName()];
+    const joinKey = this._leftField ? feature.properties[this._leftField.getName()] : undefined;
```

Comment:
Instead of adding check down here, should the method short circuit at beginning and return false if this_leftField is not defined?